### ISSUE TITLE
python deepcopy segmentation fault on version 4.24.0

### ISSUE THREAD

**Issue by criver** [protocolbuffers/protobuf#17081]
**What version of protobuf and what language are you using?**
Version: main/v3.6.0/v3.5.0 etc. (NOTE: please try updating to the latest version of protoc/runtime possible beforehand to attempt to resolve your problem)
Language: C++/Java/Python/C#/Ruby/PHP/Objective-C/Javascript

**What operating system (Linux, Windows, ...) and version?**
Ubuntu 22, OSX14.5, It happens for probouf 24.0 and above

**What runtime / compiler are you using (e.g., python version or gcc version)**
Clang 15, gcc

**What did you do?**
This error is similar to https://github.com/protocolbuffers/protobuf/issues/13485 but it happens for `deepcopy` instead.
@anandolee  

```
message Child {
  string name = 1;
}
message Parent {
  Child child = 1;
}

p = Parent()
deepcopy(p.child)
```

**What did you expect to see**
Expect it to run without error.

**What did you see instead?**

`Fatal Python error: Segmentation fault`

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

**Anything else we should know about your project / environment**

**Comment by zhangskz** [comment]
Can you confirm if you hit this even with latest 4.27.x release?

**Comment by criver** [comment]
There is no `4.27.x`, only `5.27.x`. I can't upgrade to *5* because other dependencies in my project.

**Comment by anandolee** [comment]
I believe it is the same issue with https://github.com/protocolbuffers/protobuf/issues/13485 which only happen for 4.24. 

4.23 and below, 4.25 and above should work well

**Comment by criver** [comment]
4.24 and 4.25 are broken. 4.23 works.

**Comment by criver** [comment]
@anandolee What is the plan for fixing this?

**Comment by criver** [comment]
We upgraded to 5.27 and the problem is fixed. It exists for 4.24 and 4.25.

**Comment by zhangskz** [comment]
@anandolee Was there a fix we should patch back to v4.25.x? v4.24 is out of support.
